File: SpringBoot/src/main/java/com/hospital/qlbv/entity/Role.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "roles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Role {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer roleId;
    
    @Column(name = "role_name", unique = true, nullable = false)
    private String roleName;
    
    private String description;
    
    @OneToMany(mappedBy = "role", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties("role")
    @Builder.Default
    private List<UserRole> userRoles = new ArrayList<>();
}

File: Backend/Models/Role.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class Role
{
    [Key]
    public int RoleID { get; set; }
    public string RoleName { get; set; } = string.Empty;
    public string? Description { get; set; }

    public virtual ICollection<UserRole> UserRoles { get; set; } = new List<UserRole>();
}
